File: scratch/check_duplicates.py
import json

def check_duplicates(pairs):
    d = {}
    for k, v in pairs:
        if k in d:
            print(f"Duplicate key found: {k}")
        d[k] = v
    return d

try:
    with open('apps/web/messages/ar.json', 'r', encoding='utf-8') as f:
        json.load(f, object_pairs_hook=check_duplicates)
    print("Check complete")
except Exception as e:
    print(f"Error: {e}")

File: scratch/check_duplicates.js
const fs = require('fs');

function checkDuplicateRootKeys(filePath) {
  const content = fs.readFileSync(filePath, 'utf8');
  const lines = content.split('\n');
  const rootKeys = [];
  lines.forEach((line, index) => {
    const match = line.match(/^  "([^"]+)":/);
    if (match) {
      rootKeys.push({ key: match[1], line: index + 1 });
    }
  });

  const seen = new Map();
  const duplicates = [];
  rootKeys.forEach(item => {
    if (seen.has(item.key)) {
      duplicates.push({ key: item.key, first: seen.get(item.key), second: item.line });
    } else {
      seen.set(item.key, item.line);
    }
  });

  if (duplicates.length > 0) {
    console.log(`Duplicates in ${filePath}:`);
    duplicates.forEach(d => {
      console.log(`  - Key "${d.key}" found at lines ${d.first} and ${d.second}`);
    });
  } else {
    console.log(`No duplicate root keys in ${filePath}`);
  }
}

checkDuplicateRootKeys('apps/web/messages/ar.json');
checkDuplicateRootKeys('apps/web/messages/en.json');
